README.md

```diff
@@ -1,72 +1,276 @@
+You can install the SDK directly from GitHub.
+Use a specific commit hash or branch.
 
-Then, you can install the SDK using `pip`. To do so, simply add this repo as a dependency in your `requirements.txt` file:
+**In `requirements.txt`:**
 
-```
-saltoapis-python @ git+https://github.com/saltoapis/saltoapis-python.git@<commit-hash>
+```text
+saltoapis-python @ git+https://github.com/saltoapis/saltoapis-python.git@<commit-hash-or-tag>
 ```
 
-Or run the following command:
+**Or with pip:**
 
+```sh
+pip install git+https://github.com/saltoapis/saltoapis-python.git@<commit-hash-or-tag>
 ```
-pip install git+https://github.com/saltoapis/saltoapis-python.git@<commit-hash>
-```
+
+Replace `<commit-hash-or-tag>` with a specific commit, branch, or release tag (e.g., `main` or `v1.0.0`).
 
 ## Usage
 
-Once you have installed the SDK, you can use it by importing the desired APIs.
-Additionaly, the SDK provides a simple gRPC interceptor that will automatically get and refresh valid access tokens and include them in all gRPC requests.
+Once installed, import the SDK and set up authentication.
+The SDK provides a gRPC interceptor to handle access tokens automatically.
 
-1. Create a gRPC channel:
+### Minimal example
 
 ```python
 import grpc
+from saltoapis.auth.interceptor import SaltoapisAuthInterceptor
+from saltoapis.nebula.user.v1 import user_pb2, user_pb2_grpc
 
-# create grpc channel
-channel = grpc.secure_channel(target='nebula.saltoapis.com', credentials=grpc.ssl_channel_credentials())
+# Create a secure gRPC channel
+channel = grpc.secure_channel('nebula.saltoapis.com', grpc.ssl_channel_credentials())
+
+# Set up authentication interceptor
+intercept_channel = grpc.intercept_channel(
+    channel, SaltoapisAuthInterceptor(
+        client_id='<YOUR_CLIENT_ID>',
+        client_secret='<YOUR_CLIENT_SECRET>',
+        # Replace with your actual client ID and secret
+        scopes=["openid", "offline", "profile", "email", "https://saltoapis.com/auth/nebula"]
+    )
+)
+
+# Create a service stub
+stub = user_pb2_grpc.UserServiceStub(intercept_channel)
+
+# Call an API
+response = stub.ListUsers(user_pb2.ListUsersRequest(parent="<your parent installation id>", page_size=10))
+for user in response.users:
+    print(user.display_name)
+
+# Close the channel when done
+channel.close()
 ```
 
-2. And add the saltoapis authentication interceptor:
+> **Note:** Always request the following scopes when authenticating:
+> `openid, offline, profile, email, https://saltoapis.com/auth/nebula`
+
+You can find more information about authentication at [Salto Nebula API authentication](https://developer.saltosystems.com/nebula/api/authentication/).
+
+### Further examples
+
+Here are some examples of how to use the SDK for common tasks.
+Remember to replace variables like `CLIENT_ID`, `CLIENT_SECRET` or `INSTALLATION_NAME` with real values.
+
+#### Create, update and delete user
+
+```python
+import grpc
+import random
+import string
```

Comment:
Both `import random` and `import string` can be removed.

Reply:
Done